diesel/src/mysql/connection/bind.rs

```diff
@@ -74,65 +51,124 @@ impl Binds {
     /// The caller of this function must immediately check for errors
     /// after return
     pub fn populate_dynamic_buffers(&mut self, stmt: *mut ffi::MYSQL_STMT) {
-        use std::cmp::min;
-
-        for (i, buffer) in self.data.iter_mut().enumerate() {
-            if self.errors.as_ref().unwrap()[i] != 0 {
-                let written_bytes = buffer.len();
-                let needed_capacity = self.lengths[i] as usize;
-                buffer.reserve(needed_capacity - written_bytes);
-                self.mysql_binds[i].buffer = buffer.as_mut_ptr() as *mut libc::c_void;
-                mem::swap(&mut self.lengths[i], &mut self.mysql_binds[i].buffer_length);
+        for (i, data) in self.data.iter_mut().enumerate() {
+            let bind = &mut self.mysql_binds[i];
```

Comment:
This could also be written as `for (i, (data, bind))` in self.data.iter_mut().zip(&mut self.mysql_binds).enumerate()`. Thoughts on which is nicer?